src/evict.c

```diff
@@ -402,7 +402,11 @@ int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *lev
         if (level) *level = 0;
         return C_OK;
     }
-    if (mem_reported <= server.maxmemory && !level) return C_OK;
+
+    if (server.maxmemory_reserved_scale) {
+        if (mem_reported <= server.maxmemory_available && !level) return C_OK;
+    } else if (mem_reported <= server.maxmemory && !level)
+        return C_OK;
```

Comment:
nit. add {} to match the `if` part?
```suggestion
    } else if (mem_reported <= server.maxmemory && !level) {
        return C_OK;
    }
```

Reply:
Address.